theme-utils.mjs

```diff
@@ -29,6 +29,591 @@ function showHelp(){
+
+	const result = await executeOnSandbox(`
+		cd ${sandboxPublicThemesFolder};
+		echo '${commitMessage}' > ${commitTempFileLocation}
+		svn add --force * --auto-props --parents --depth infinity -q
+		arc diff --create --message-file ${commitTempFileLocation}
+	`);
+
+	const phabricatorUrl = getPhabricatorUrlFromResponse(result);
+
+	console.log('Diff Created at: ', phabricatorUrl);
+
+	if(phabricatorUrl) {
+		open(phabricatorUrl);
+	}
+
+	return phabricatorUrl;
+}
+
+/*
+ Utility to pull the Phabricator URL from the diff creation command.
+ Used by createGitPhabricatorDiff
+*/
+function getPhabricatorUrlFromResponse(response){
+	return response 
+		?.split('\n')
+		?.find( item => {
+			return item.includes('Revision URI: ');
+		})
+		?.split("Revision URI: ")[1];
+}
+
+/*
+ Create a git tag at the current hash.
+ In the description include the commit logs since the given hash.
+ Include the (cleansed) Phabricator link.
+*/
+async function tagDeployment(options={}) {
+
+	let hash = options.hash || await getLastDeployedHash();
+
+	let workInTheOpenPhabricatorUrl = '';
+	if (options.diffUrl) {
+		workInTheOpenPhabricatorUrl = `Phabricator: ${options.diffUrl.split('a8c.com/')[1]}-code`;
+	}
+	let projectVersion = await executeCommand(`node -p "require('./package.json').version"`);
+	let logs = await executeCommand(`git log --reverse --pretty=format:%s ${hash}..HEAD`);
+	let tag = `v${projectVersion}`;
+	let message = `Deploy Themes ${tag} to wpcom. \n\n${logs} \n\n${workInTheOpenPhabricatorUrl}`;
+
+	await executeCommand(`
+		git tag -a ${tag} -m "${message}"
+		git push origin ${tag}
+	`);
+}
+
+/*
+ Execute a command on the sandbox.
+ Expects the following to be configured in your ~/.ssh/config file:
+
+Host wpcom-sandbox
+	User wpdev
+	HostName SANDBOXURL.wordpress.com
+	ForwardAgent yes
+*/
+function executeOnSandbox(command){
+	return executeCommand(`ssh -TA ${remoteSSH} << EOF 
+${command} 
+EOF`);
+}
+
+/*
+ Execute a command locally.
+*/
+async function executeCommand(command) {
+	return new Promise((resolove, reject) => {
+
+		let child;
+		let response = '';
+		let errResponse = '';
+
+		if (isWin) {
+			child = spawn('cmd.exe', ['/s', '/c', '"' + command + '"'], {
+				windowsVerbatimArguments: true,
+				stdio: [process.stdin, 'pipe', 'pipe'],
+			})
+		} else {
+			child = spawn(process.env.SHELL, ['-c', command]);
+		}
+
+		child.stdout.on('data', (data) => {
+			response += data;
+		});
+
+		child.stderr.on('data', (data) => {
+			errResponse += data;
+		});
+
+		child.on('exit', (code) => {
+			if (code !== 0) {
+				reject(errResponse.trim());
+			}
+			resolove(response.trim());
+		});
+	});
+}
+import { spawn } from 'child_process';
+import fs from 'fs';
+import open from 'open';
+
+const remoteSSH = 'wpcom-sandbox';
+const sandboxPublicThemesFolder = '/home/wpdev/public_html/wp-content/themes/pub';
+const sandboxRootFolder = '/home/wpdev/public_html/';
+const isWin = process.platform === 'win32';
+
+(async function start() {
```

Comment:
this whole block seems to be in the file twice?

Reply:
I had a problem yesterday where the entire file was duplicated.  I don't see that now though and thought I fixed and pushed that yesterday.